cpp/oneapi/dal/table/backend/interop/sycl_table_adapter_dpc.cpp

```diff
@@ -0,0 +1,289 @@
+    return convert_exception_to_status([&]() {
+        return read_column_values_impl(feature_idx, vector_idx, value_num, rwflag, block);
+    });
+}
+
+auto sycl_table_adapter::releaseBlockOfRows(block_desc_t<double>& block) -> status_t {
+    block.reset();
+    return status_t();
+}
+
+auto sycl_table_adapter::releaseBlockOfRows(block_desc_t<float>& block) -> status_t {
+    block.reset();
+    return status_t();
+}
+
+auto sycl_table_adapter::releaseBlockOfRows(block_desc_t<int>& block) -> status_t {
+    block.reset();
+    return status_t();
+}
+
+auto sycl_table_adapter::releaseBlockOfColumnValues(block_desc_t<double>& block) -> status_t {
+    block.reset();
+    return status_t();
+}
+
+auto sycl_table_adapter::releaseBlockOfColumnValues(block_desc_t<float>& block) -> status_t {
+    block.reset();
+    return status_t();
+}
+
+auto sycl_table_adapter::releaseBlockOfColumnValues(block_desc_t<int>& block) -> status_t {
+    block.reset();
+    return status_t();
+}
+
+auto sycl_table_adapter::assign(float) -> status_t {
+    return daal::services::ErrorMethodNotImplemented;
+}
+
+auto sycl_table_adapter::assign(double) -> status_t {
+    return daal::services::ErrorMethodNotImplemented;
+}
+
+auto sycl_table_adapter::assign(int) -> status_t {
+    return daal::services::ErrorMethodNotImplemented;
+}
+
+auto sycl_table_adapter::allocateDataMemoryImpl(daal::MemType) -> status_t {
+    return daal::services::ErrorMethodNotImplemented;
+}
+
+auto sycl_table_adapter::setNumberOfColumnsImpl(std::size_t) -> status_t {
+    return daal::services::ErrorMethodNotImplemented;
+}
+
+int sycl_table_adapter::getSerializationTag() const {
+    ONEDAL_ASSERT(!"sycl_table_adapter: getSerializationTag() is not implemented");
+    return -1;
+}
+
+auto sycl_table_adapter::serializeImpl(daal_dm::InputDataArchive* arch) -> status_t {
+    return daal::services::ErrorMethodNotImplemented;
+}
+
+auto sycl_table_adapter::deserializeImpl(const daal_dm::OutputDataArchive* arch) -> status_t {
+    return daal::services::ErrorMethodNotImplemented;
+}
+
+void sycl_table_adapter::freeDataMemoryImpl() {
+    base::freeDataMemoryImpl();
+    original_table_ = homogen_table{};
+}
+
+template <typename BlockData>
+auto sycl_table_adapter::read_rows_impl(std::size_t vector_idx,
+                                        std::size_t vector_num,
+                                        rw_mode_t rwflag,
+                                        block_desc_t<BlockData>& block) -> status_t {
+    if (rwflag != daal_dm::readOnly) {
+        ONEDAL_ASSERT(!"Data is accessible in read-only mode by design");
+        return daal::services::ErrorMethodNotImplemented;
+    }
+
+    const block_info info{ block, vector_idx, vector_num };
+    if (!check_row_indexes_in_range(info)) {
+        return daal::services::ErrorIncorrectIndex;
+    }
+
+    const auto [buffer, status] = pull_rows_buffer<BlockData>(info);
```

Comment:
What if the `block` already contains the buffer where we can write?
I'm not sure we need to apply such optimizations here since this adapter is temporary solution

Reply:
`SyclHomogenTable` has similar behavior